x/subkeys/doc.go

```diff
@@ -0,0 +1,32 @@
+/*
+Package subkeys provides functionality for delegating sub-permissions
+from one account (key) to another account (key).
+
+The first implementation allows the delegation for the payment of transaction fees.
+Effectively, this allows for a user to pay fees using the balance of an account
+different from their own. Example use cases would be allowing a key on a device to
+pay for fees using a master wallet, or a third party service allowing users to
+pay for transactions without ever really holding their own tokens. This package
+provides ways for specifying fee allowances such that delegating fees
+to another account can be done with clear and safe restrictions.
+
+A user would delegate fees to a user using MsgDelegateFeeAllowance and revoke
+that delegation using MsgRevokeFeeAllowance. In both cases Granter is the one
+who is delegating fees and Grantee is the one who is receiving the delegation.
+So grantee would correspond to the one who is signing a transaction and the
+granter would be the address they place in StdTx.FeeAccount.
+
+The fee allowance that a grantee receives is specified by an implementation of
+the FeeAllowance interface. Two FeeAllowance implementations are provided in
+this package: BasicFeeAllowance and PeriodicFeeAllowance.
+
+TODO: update if needed with new modular ante handler
+
+In order to integrate this into an application, the "ante handler" which deducts
+fees must call Keeper.AllowDelegatedFees to check if
+the provided StdTx.Fee can be delegated from the Std.TxFeeAccount address
+to the first signer of the transaction. An example usage would be:
+
+allow := feeDelegationKeeper.AllowDelegatedFees(ctx, signers[0], stdTx.FeeAccount, stdTx.Fee.Amount)
+*/
+package subkeys
```

Comment:
What is the logic for packaging both fee delegation and msg delegation into one module? These functionalities do not depend on each so if there are no functional interdependencies why bundle them together? They are orthogonal, separation of concerns seems like the more logical approach.

And why was `subkeys` chosen as the preferred name? `delegation` seems more generic whereas `subkeys` to me has the implication that I am delegated permissions from one of my keys to another one of my keys. `delegation` implies I can delegate actions of fees to any other account, not just ones I own. For instance, I may delegate `gov` permissions to some other person - they are not a "subkey" of my account, they are more like my representative.

Reply:
I need to make modifications to the whole ante stack for both. (This is the new modular ante handler)

I would really like to have one ante package in auth and one in (module to be named), and not fragments in a few different modules that have to be assembled. 

I actually like the idea of plugging in msg delegation to the permission checking ante-handler.

Reply:
Cleaned up docs in 5f9e391de